core/src/main/scala/cromwell/util/DatabaseUtil.scala

```diff
@@ -0,0 +1,22 @@
+package cromwell.util
+
+import java.sql.SQLTransientException
+
+import akka.actor.ActorSystem
+import cromwell.core.retry.{Retry, SimpleExponentialBackoff}
+
+import scala.concurrent.Future
+import scala.concurrent.duration._
+import scala.language.postfixOps
+
+object DatabaseUtil {
+  private def isTransient(throwable: Throwable): Boolean = throwable match {
+    case _: SQLTransientException => true
+    case _ => false
+  }
+
+  def withRetry[A](f: => Future[A])(implicit actorSystem: ActorSystem): Future[A] = {
```

Comment:
Even if I pass `() => f` to the `Retry.withRetry` on line 20 ?
Anyway I can change it back to a full function I just find it more practical to use this way, but if doesn't work then probably not worth it :p